server-protocol.md

```diff
@@ -104,6 +124,55 @@ was sent in the wrong order), an `error` response will be sent, This response
 will include the error string in the `error` key, and a full copy of the
 original message dictionary in `orig`.
 
+
+## Permission to Use the Server
+
+Server operators may wish to deny service to some clients. We
+generally refer to this as "Permission" and imagine future additions
+of use-cases and methods, although only one such pair is decribed
+currently.
+
+One such use-case is if the server is under a Denial of Service (DoS)
+attack or other malicious activity. We describe a "proof of work"
+scheme for clients to gain permission to use a server under DoS
+attack.
+
+In the `welcome` message the server may include a `permission-required` key. If provided, this will point to a `dict` with one key per supported method along with any method-specific options.
+For example:
+
+    {
+        "none": {}
+        "hashcash": {
+            "bits": 6,
+            "resource": "see description"
+        }
+    }
+
+Currently, the following are supported by the protocol:
+
+* `none`: no permission required, send a normal `bind`.
+* `hashcash`: Includes keys for `bits` and `resource` used for input as
+   per the [Hashcash](http://hashcash.org) specifications. The resource
+   string is a web-safe base64-encoded token unique to this session
+   (although the client shouldn't depend on that). The client must
```

Comment:
Why not make this an opaque string for the client, so that it is implementation defined on the server side?

Reply:
okay, i can take it out? (Mostly I guess that's a note about how the Python server will work .. but yeah clients shouldn't depend on anything in it and need to simply use it)

Reply:
okay, re-worded further.